File: frontend/src/services/SellerService.ts
// src/services/SellerService.ts

import api from './api';

export interface SellerStartRequest {
  storeName: string;
  email: string;
  phoneNum: string;
  address: string;
  userId: string; // UUID string
}

export interface SellerShippingRequest {
  express?: boolean;
  standard?: boolean;
  economy?: boolean;
}

export interface SellerCreateResponse {
  sellerId: string;
  storeName: string;
  email: string;
  phoneNum: string;
  address: string;
  shippingConfigured?: boolean;
}
export async function startRegistration(data: SellerStartRequest) {
  return await api.post<SellerCreateResponse>('/api/sellers', data);
}
export async function configureShipping(
  sellerId: string,
  data: SellerShippingRequest,
) {
  return await api.post<SellerCreateResponse>(
    `/api/sellers/${sellerId}/shipping`,
    data,
  );
}

// Analytics types
export interface DailyRevenue {
  date: string;
  revenue: number;
  orderCount: number;
}

export interface AnalyticsSummary {
  totalRevenue: number;
  totalOrders: number;
  pendingOrders: number;
  completedOrders: number;
  revenueChart: DailyRevenue[];
}

/**
 * Get analytics summary for the current seller
 */
export async function getAnalyticsSummary(): Promise<AnalyticsSummary> {
  const response = await api.get<AnalyticsSummary>('seller/analytics/summary');
  return response;
}

File: backend/src/main/java/Bazaar/com/project/feature/User/_Seller/service/SellerService.java
package Bazaar.com.project.feature.User._Seller.service;

import java.util.UUID;

import Bazaar.com.project.feature.User._Seller.dto.SellerShippingRequest;
import Bazaar.com.project.feature.User._Seller.dto.SellerStartRequest;
import Bazaar.com.project.feature.User.model.Seller;

public interface SellerService {
    Seller startRegistration(SellerStartRequest request);

    Seller configureShipping(UUID sellerId, SellerShippingRequest request);
}
